internal/service/advancedcluster/model_advanced_cluster_test.go

```diff
@@ -21,6 +23,128 @@ var (
+		testSchema = map[string]*schema.Schema{
+			"project_id": {Type: schema.TypeString},
+		}
+		tf1SameIDSameZone = map[string]any{
+			"id":             expectedID,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTf},
+			"zone_name":      expectedZoneName,
+		}
+		tf2NoIDSameZone = map[string]any{
+			"id":             nil,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTf},
+			"zone_name":      expectedZoneName,
+		}
+		tf3NoIDDiffZone = map[string]any{
+			"id":             nil,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTfDiffZone},
+			"zone_name":      unexpectedZoneName,
+		}
+		tf4diffIDDiffZone = map[string]any{
+			"id":             "unique",
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTfDiffZone},
+			"zone_name":      unexpectedZoneName,
+		}
+	)
+	testCases := map[string]struct {
+		adminSpecs   []admin.ReplicationSpec
+		tfInputSpecs []any
+		expectedLen  int
+	}{
+		"empty admin spec should return empty list": {
+			[]admin.ReplicationSpec{},
+			[]any{tf1SameIDSameZone},
+			0,
+		},
+		"existing id, should match admin": {
+			[]admin.ReplicationSpec{admin1},
+			[]any{tf1SameIDSameZone},
+			1,
+		},
+		"existing different id, should change to admin spec": {
+			[]admin.ReplicationSpec{admin1},
+			[]any{tf4diffIDDiffZone},
+			1,
+		},
+		"missing id, should be set when zone_name matches": {
+			[]admin.ReplicationSpec{admin1},
+			[]any{tf2NoIDSameZone},
+			1,
+		},
+		"missing id and diff zone, should change to admin spec": {
+			[]admin.ReplicationSpec{admin1},
+			[]any{tf3NoIDDiffZone},
+			1,
+		},
+		"existing id, should match correct api spec using `id` and extra api spec added": {
+			[]admin.ReplicationSpec{admin2, admin1},
+			[]any{tf1SameIDSameZone},
+			2,
+		},
+		"missing id, should match correct api spec using `zone_name` and extra api spec added": {
+			[]admin.ReplicationSpec{admin2, admin1},
+			[]any{tf2NoIDSameZone},
+			2,
+		},
+		"two matching specs should be set to api specs": {
+			[]admin.ReplicationSpec{admin1, admin2},
+			[]any{tf1SameIDSameZone, tf4diffIDDiffZone},
+			2,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			peeringAPI := mockadmin.NetworkPeeringApi{}
+
+			peeringAPI.EXPECT().ListPeeringContainerByCloudProviderWithParams(mock.Anything, mock.Anything).Return(admin.ListPeeringContainerByCloudProviderApiRequest{ApiService: &peeringAPI})
+			peeringAPI.EXPECT().ListPeeringContainerByCloudProviderExecute(mock.Anything).Return(&admin.PaginatedCloudProviderContainer{Results: &[]admin.CloudProviderContainer{{Id: conversion.StringPtr("c1"), RegionName: &regionName, ProviderName: &providerName}}}, nil, nil)
+
+			client := &admin.APIClient{
+				NetworkPeeringApi: &peeringAPI,
+			}
+			resourceData := schema.TestResourceDataRaw(t, testSchema, map[string]any{"project_id": "p1"})
```

Comment:
Helper to create `schema.ResourceData` needed for the `project_id` used to find `ContainerIds`